cmd/migrator/cmd/catalog.go

```diff
@@ -0,0 +1,74 @@
+package cmd
+
+import (
+	"context"
+	"io"
+
+	"github.com/gomods/athens/pkg/storage"
+)
+
+// returns the info bytes, mod, and zip (in that order)
+func getCompleteMod(
+	ctx context.Context,
+	getter storage.Getter,
+	mod string,
+	ver string,
+) ([]byte, []byte, io.ReadCloser, error) {
+	infoBytes, err := getter.Info(ctx, mod, ver)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	modBytes, err := getter.GoMod(ctx, mod, ver)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	zip, err := getter.Zip(ctx, mod, ver)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return infoBytes, modBytes, zip, nil
+}
+
+func transfer(
+	ctx context.Context,
+	cataloger storage.Cataloger,
+	from storage.Getter,
+	to storage.Saver,
+) error {
+	token := ""
+	// TODO: parallelize this
+	for {
+		pathParams, newToken, err := cataloger.Catalog(ctx, token, 100)
+		if err != nil {
+			return err
+		}
+		for _, pathParam := range pathParams {
```

Comment:
moving storages is likely a very timely process, we should do a couple of things: 

1. Parallelize reads/writes 
2. Check if the destination storage has the module@version first,  instead of always overriding it. 
3. Further enhancement: make the migration process resumable, though with 2 it should be good enough i think.

Reply:
@marwan-at-work for (1), I [agree that it should be parallelized ](https://github.com/gomods/athens/pull/1609/files#diff-ef0e160aad5c050733a77e59a54b3807R39). I was waiting to see if the general structure of this looks ok. 

If you're ok with [my suggestion for a follow-up](https://github.com/gomods/athens/pull/1609#issuecomment-627635998), I can continue on with parallelizing and doing reads before writes in here. Let me know ✌️